src/plugins/plugin_manager.h

```diff
@@ -4,16 +4,15 @@
 #include "plugins/plugin.h"
 
 #include <memory>
-#include <string_view>
 #include <vector>
 
 namespace NOX {
 
 class Plugin;
 
-class PluginManager : public Singleton<PluginManager> {
+class PluginManagerImpl {
   public:
-    [[nodiscard]] bool load(std::string_view name, PluginFilenameCreationStrategy createFilename);
+    [[nodiscard]] bool registerPlugin(std::unique_ptr<Plugin> &&plugin);
 
   private:
     [[nodiscard]] bool isLoaded(std::string_view name) const;
```

Comment:
does PluginManagerImpl has to be in .h ? isn't it implementation detail and couldn't it be moved to cpp?

Reply:
Turns out it wasn't even needed. Deleted it.